# Week 8 Independent Lab
### University of Montana MSBA Program | Spring 2025
### BGEN 632 | Alex McKenzie

In [6]:
import pandas as pd

path = '/Users/AlexMcKenzie/Library/CloudStorage/GoogleDrive-alex.c.mckenzie@gmail.com/My Drive/Education/UM Graduate School/BGEN632/week8labs/data/gh_turnover_four_projects.csv'
turnover_df = pd.read_csv(path)

# to verify it loaded correctly
turnover_df.head()

project_id domain language  windows  window_idx  num_team  num_commits  \
0        1101  OTHER   Python       23          22        26        22688   
1        1101  OTHER   Python       23          22        26        22688   
2        1101  OTHER   Python       23          22        26        22688   
3        1101  OTHER   Python       23          22        26        22688   
4        1101  OTHER   Python       23          22        26        22688   

   blau_gender  Gini_gh_ten  user_id  Core1  commits  propCommits gender  \
0          0.0     0.422259  1614129      1    22138     0.975758   male   
1          0.0     0.422259  1614101      0      124     0.005465   male   
2          0.0     0.422259    94412      0       53     0.002336   male   
3          0.0     0.422259    88459      0       49     0.002160   male   
4          0.0     0.422259   738787      0       45     0.001983   male   

   github_tenure  leavesNextQ  
0            426            0  
1             46            0  
2            941            1  
3           1870            0  
4            781            0

In [18]:
turnover_df = turnover_df.rename(columns={
    'domain':'project_domain',
    'language':'project_language',
    'windows':'project_age',
    'window_idx':'quarter',
    'num_team':'team_size',
    'num_commits':'project_commits',
    'blau_gender':'gender_blau',
    'Gini_gh_ten':'gh_tenure_gini',
    'Core1':'core_dev',
    'commits':'user_commits',
    'propCommits':'commit_proportion',
    'github_tenure':'gh_tenure',
    'leavesNextQ':'leaves_next_q'
})

In [19]:
turnover_df['user_commits'].mean()

339.36575052854124

In [20]:
turnover_df['user_commits'].median()

18.0

In [21]:
turnover_new = new_df.filter(regex='(^p)|(g)')
turnover_new.columns.tolist()

['project_id',
 'project_domain',
 'project_language',
 'project_age',
 'project_commits',
 'gender_blau',
 'gh_tenure_gini',
 'gender',
 'gh_tenure']

In [22]:
turnover_new_filtered = turnover_new[turnover_new['project_commits'] > 9000]
turnover_new_filtered

project_id project_domain project_language  project_age  project_commits  \
0          1101          OTHER           Python           23            22688   
1          1101          OTHER           Python           23            22688   
2          1101          OTHER           Python           23            22688   
3          1101          OTHER           Python           23            22688   
4          1101          OTHER           Python           23            22688   
..          ...            ...              ...          ...              ...   
116        1101          OTHER           Python           23            14210   
117        1101          OTHER           Python           23            14210   
118        1101          OTHER           Python           23            14210   
119        1101          OTHER           Python           23            14210   
120        1101          OTHER           Python           23            14210   

     gender_blau  gh_tenure_gini gender  gh_tenure  
0            0.0        0.422259   male        426  
1            0.0        0.422259   male         46  
2            0.0        0.422259   male        941  
3            0.0        0.422259   male       1870  
4            0.0        0.422259   male        781  
..           ...             ...    ...        ...  
116          0.0        0.411372   male        495  
117          0.0        0.411372   male        487  
118          0.0        0.411372   male        317  
119          0.0        0.411372   male        418  
120          0.0        0.411372   male        179  

[114 rows x 9 columns]

In [23]:
turnover_newer = (
    new_df
    .filter(regex='(^p)|(g)')
    .loc[lambda df: df['project_commits'] > 9000]
)
turnover_newer

project_id project_domain project_language  project_age  project_commits  \
0          1101          OTHER           Python           23            22688   
1          1101          OTHER           Python           23            22688   
2          1101          OTHER           Python           23            22688   
3          1101          OTHER           Python           23            22688   
4          1101          OTHER           Python           23            22688   
..          ...            ...              ...          ...              ...   
116        1101          OTHER           Python           23            14210   
117        1101          OTHER           Python           23            14210   
118        1101          OTHER           Python           23            14210   
119        1101          OTHER           Python           23            14210   
120        1101          OTHER           Python           23            14210   

     gender_blau  gh_tenure_gini gender  gh_tenure  
0            0.0        0.422259   male        426  
1            0.0        0.422259   male         46  
2            0.0        0.422259   male        941  
3            0.0        0.422259   male       1870  
4            0.0        0.422259   male        781  
..           ...             ...    ...        ...  
116          0.0        0.411372   male        495  
117          0.0        0.411372   male        487  
118          0.0        0.411372   male        317  
119          0.0        0.411372   male        418  
120          0.0        0.411372   male        179  

[114 rows x 9 columns]

In [28]:
filtered_df = (
    turnover_df
    .filter(regex='(e$)|s')
    .loc[lambda df: df['gh_tenure'] >= 1991]
)

In [29]:
mean_filtered = filtered_df['user_commits'].mean()
median_filtered = filtered_df['user_commits'].median()

mean_filtered, median_filtered

(3.5, 3.5)

#### Data Analysis:

Original mean calculation: 339.36
Original median calculation: 18.0

Filtered mean calculation: 3.5
Filtered median calculation 3.5

# References and Notes:

I used ChatGPT for this assigment:

https://drive.google.com/file/d/1vMMt8pYgaSdo5JP3CnZks_okEHH5Uwma/view?usp=sharing

While I made a serious effort the last couple weeks to demonstrate my efforts to get some solid coding done without ChatGPT, I wasn't able to put forth the same this week due to professional/familiar constraints. While the Instructor-Led assignment was uniquely challenging (see comments in that submission), by the time I got to this assignment I unfortunately had to just get it off my desk. Please review my ChatGPT transcript and grade this assignment accordingly. Thank you! -Alex